charts/vector/templates/_pod.tpl

```diff
@@ -172,6 +172,10 @@ affinity:
 tolerations:
 {{ toYaml . | indent 2 }}
 {{- end }}
+{{- with  .Values.topologySpreadConstraints }}
+topologySpreadConstraints:
+{{- toYaml . | nindent 2 }}
+{{- end }}
```

Comment:
```suggestion
{{- if  .Values.topologySpreadConstraints }}
topologySpreadConstraints:
  labelSelector:
    matchLabels:
{{ include "vector.selectorLabels" . | nindent 6 }}
{{- toYaml .Values.topologySpreadConstraints | nindent 2 }}
{{- end }}
```

Not perfect, but this allows you to just pass in key/values of the constraints and re-use the selectorLabels.